eliot-service/eliot/symbolicate_resource.py

```diff
@@ -149,6 +155,58 @@ def validate_stacks(stacks, modules):
                 )
 
 
+def bytes_split_generator(item, sep):
+    """Takes a bytes or bytearray and returns a generator of parts split on sep
+
+    :arg item: bytes-like object
+    :arg sep: bytes-like object
+
+    :returns: generator splitting the item
+
+    """
+    index = 0
+    len_item = len(item)
+
+    while index <= len_item:
+        next_index = item.find(sep, index)
+        if next_index == -1:
+            break
+
+        yield item[index:next_index]
+        index = next_index + len(sep)
+
+
+def get_module_filename(sym_file, debug_filename):
+    """Returns the module filename
+
+    On Windows, this will be the pe_file in the INFO line:
+
+        INFO CODE_ID xxx pe_file
+
+    On other platforms, this is the debug_filename.
+
+    :arg bytes sym_file: the sym file as bytes
+    :arg str debug_filename: the debug filename
+
+    :returns: the module filename
+
+    """
+    # Iterate through the first few lines of the file until we hit FILE in which
+    # case there's no INFO for some reason or we hit the first INFO.
+    for line in bytes_split_generator(sym_file, b"\n"):
+        if line.startswith(b"INFO"):
```

Comment:
I'm basing this structure on what dump_syms outputs where INFO is close to the top of the file.